### PR TITLE
Add batch processor feature document

### PR DESCRIPTION
**Pull request by RyanLassigBanks**
Signed-off-by: Ryan Banks <rbanks@bitwise.io>

### CODE REVIEW

File: community/planning/batch_processor.md [hunk 1]
```diff
@@ -0,0 +1,84 @@
+<!--
+  Copyright 2021 Cargill Incorporated
+  Licensed under Creative Commons Attribution 4.0 International License
+  https://creativecommons.org/licenses/by/4.0/
+-->
+
+# Batch Processor
```
Comment (vaporos) [inline community/planning/batch_processor.md:7]: Rename to Batch Submitter (at least for now); the current component that is called Batch Submitter can be renamed. "Batch Processor" is too close to "Transaction Processor".

File: community/planning/batch_processor.md [hunk 2]
```diff
@@ -0,0 +1,84 @@
...
+
+The batch processor is a proposed component meant to monitor a database that
+holds batched transactions and send them to either a Hyperledger Sawtooth or
+Splinter network. Batch processors are meant to be run as part of a cluster with
+other batch processors in order to provide redundancy or reduce the time it
+takes for batches to be processed.
+
+## Batch Submission Algorithm
```
Comment (vaporos) [inline community/planning/batch_processor.md:15]: This algorithm, without some form of locking, would seem to suffer from multiple processes easily claiming the same batches due to the race condition between the SELECT and implied UPDATE. Thus more detail here around the specific mechanics would be useful.

File: community/planning/batch_processor.md [hunk 3]
```diff
@@ -0,0 +1,84 @@
...
+up and performing the following actions.
+
+1. A processor will query the database for a configurable number of unclaimed
+batches (the `claim_limit`). An unclaimed batch is a batch whose submitted
+status is `false`, and its `claim_expires` time stamp is `NULL` or less than the
+`CURRENT_TIMESTAMP` of the database.
+```
+SELECT * FROM batches
+  WHERE submitted = false
+    AND (claim_expirers is NULL
```
Comment (vaporos) [inline community/planning/batch_processor.md:27]: Typo - meant to be claim_expires.

File: community/planning/batch_processor.md [hunk 4]
```diff
@@ -0,0 +1,84 @@
...
+      OR claim_expires < CURRENT_TIMESTAMP) LIMIT {claim_limit};
+```
+
+2. The processor will then update the `claim_expires` column to lay claim to
```
Comment (vaporos) [inline community/planning/batch_processor.md:31]: What are reasonable values for claim_expires and the sleep interval?

**Review by vaporos — CHANGES_REQUESTED** [review]

File: community/planning/batch_processor.md (hunk 4 above)
Comment (RyanLassigBanks) [reply community/planning/batch_processor.md:31]: The current default is 10 seconds.

File: community/planning/batch_processor.md (hunk 1 above)
Comment (RyanLassigBanks) [reply community/planning/batch_processor.md]: renamed to batch submitter

File: community/planning/batch_processor.md (hunk 2 above)
Comment (RyanLassigBanks) [reply community/planning/batch_processor.md:15]: Updated to specify that a submitter is claiming a lock and that the update and query for batches are a part of the same transaction.

File: community/planning/batch_processor.md (hunk 3 above)
Comment (RyanLassigBanks) [reply community/planning/batch_processor.md]: updated

**Review by dplumb94 — APPROVED** [review]

File: community/planning/features.md [hunk 5]
```diff
@@ -18,3 +18,4 @@ community meeting.
 | Feature | Description |
 | ------- | ----------- |
 | [Location Sapling]({% link community/planning/location_sapling.md %}) | A front-end for Grid Location functionality |
+| [Batch Processor]({% link community/planning/batch_processor.md %}) | A daemon processor for submitting batches |
```
Comment (vaporos) [inline community/planning/features.md:21]: Should rename the file - batch_submitter.md

Comment (RyanLassigBanks) [reply community/planning/features.md]: renamed to batch_submitter.md

**Review by dplumb94 — APPROVED** [review]

**Pull request merged**